src/Datadog.Trace.ClrProfiler.Managed/Integrations/HttpMessageHandlerIntegration.cs

```diff
@@ -293,5 +307,85 @@ private static bool IsTracingEnabled(object headers)
 
             return true;
         }
+
+        /********************
+         * Duck Typing Types
+         */
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+#pragma warning disable SA1201 // Elements must appear in the correct order
+#pragma warning disable SA1600 // Elements must be documented
+
+        [DuckCopy]
+        public struct HttpRequestMessageStruct
+        {
+            public HttpMethodStruct Method;
+
+            public Uri RequestUri;
+
+            public IRequestHeaders Headers;
+        }
+
+        [DuckCopy]
+        public struct HttpMethodStruct
+        {
+            public string Method;
+        }
+
+        public interface IRequestHeaders
+        {
+            bool TryGetValues(string name, out IEnumerable<string> values);
+
+            bool Remove(string name);
+
+            void Add(string name, string value);
+        }
+
+        [DuckCopy]
+        public struct HttpResponseMessageStruct
+        {
+            public int StatusCode;
+        }
+
+        [DuckCopy]
+        public struct TaskObjectStruct
+        {
+            public object Result;
+        }
```

Comment:
I think we'll need this struct in many integrations.

What is best from a performance perspective? Redeclare it every time or reuse the same for all integrations?

Reply:
Yes, If we want to kick the fast path always it's better to redeclare this struct. That will avoid a lookup to the `ConcurrentDictionary`